src/kernel/include/shared/cos_types.h

```diff
@@ -840,8 +848,8 @@ static inline void
 cos_mem_fence(void)
 { __asm__ __volatile__("mfence" ::: "memory"); }
 
-// ncpu * 16 (or max 256) entries. can be increased if necessary.
-#define COS_THD_INIT_REGION_SIZE (((NUM_CPU*16) > (1<<8)) ? (1<<8) : (NUM_CPU*16))
+/* 256 entries. can be increased if necessary */
+#define COS_THD_INIT_REGION_SIZE (1<<8)
```

Comment:
Did you mean to remove all of the `NUM_CPU` logic here?  Should we vary this based on the # of cores?